coordinator/src/main/java/org/apache/uniffle/coordinator/SelectStorageStrategy.java

```diff
@@ -17,24 +17,14 @@
 
 package org.apache.uniffle.coordinator;
 
+import java.util.List;
 import java.util.Map;
 
-import org.apache.uniffle.common.RemoteStorageInfo;
 import org.apache.uniffle.coordinator.LowestIOSampleCostSelectStorageStrategy.RankValue;
 
 public interface SelectStorageStrategy {
 
-  RemoteStorageInfo pickRemoteStorage(String appId);
+  List<Map.Entry<String, RankValue>> detectStorage(String uri);
 
-  void incRemoteStorageCounter(String remoteStoragePath);
-
-  void decRemoteStorageCounter(String storagePath);
-
-  void removePathFromCounter(String storagePath);
-
-  Map<String, RemoteStorageInfo> getAvailableRemoteStorageInfo();
-
-  Map<String, RemoteStorageInfo> getAppIdToRemoteStorageInfo();
-
-  Map<String, RankValue> getRemoteStoragePathRankValue();
+  String pickStorage(List<Map.Entry<String, RankValue>> uris, String appId);
```

Comment:
Should we return `RemoteStorageInfo`?

Reply:
No need, because the return value of `pickStorage` only needs to be passed to `incRemoteStorageCounter`, `incRemoteStorageCounter` only needs `String` type, and `String` type is more general than `RemoteStorageInfo`.

Reply:
> No need, because the return value of `pickStorage` only needs to be passed to `incRemoteStorageCounter`, `incRemoteStorageCounter` only needs `String` type, and `String` type is more general than `RemoteStorageInfo`.

`String` is too general to describe what we want to return. `RemoteStorageInfo` is more accurate.

Reply:
OK, updated.